src/components/subscriptions/SubscriptionDetailPage.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import SubscriptionExpirationBanner from './expiration/SubscriptionExpirationBanner';
+import SubscriptionExpirationModal from './expiration/SubscriptionExpirationModal';
+import SubscriptionDetails from './SubscriptionDetails';
+import LicenseAllocationDetails from './licenses/LicenseAllocationDetails';
+import { SubscriptionContext } from './SubscriptionData';
+import SubscriptionDetailContextProvider from './SubscriptionDetailContextProvider';
+import { NotFound } from '../NotFoundPage';
+
+const SubscriptionDetailPage = ({ match }) => {
+  const { params: { subscriptionUUID } } = match;
+  const { data: subscriptions } = useContext(SubscriptionContext);
+  if (!subscriptions?.count) {
+    return (
+      <NotFound />
```

Comment:
unrelated random thought: I have been thinking that at some point we should enhance the messaging of this page or add more similar pages to indicate better what the cause of the 404 page was, but not for this PR!